spec-tests/src/test/java/org/candlepin/spec/products/ProductResourceSpecTest.java

```diff
@@ -0,0 +1,241 @@
+        }
+
+        // Verify we get all owners with a product id from each owner
+        List<OwnerDTO> owners2 = productsApi.getProductOwners(productIdFromAllOwners);
+        assertEquals(3, owners2.size());
+
+        // Unknown product
+        List<OwnerDTO> owners = productsApi.getProductOwners(Arrays.asList("UnknownProductId"));
+        assertEquals(0, owners.size());
+    }
+
+    @Test
+    @DisplayName("refreshes pools for orgs owning products")
+    public void refreshPoolsForOrgsOwningProducts() throws Exception {
+        if (isStandalone()) {
+            return;
+        }
+
+        OwnerDTO owner1 = ownerApi.createOwner(Owners.random());
+        OwnerDTO owner2 = ownerApi.createOwner(Owners.random());
+        OwnerDTO owner3 = ownerApi.createOwner(Owners.random());
+        Map<OwnerDTO, Collection<ProductDTO>> ownerToProducts
+            = setupOrgProductsAndPools(Arrays.asList(owner1, owner2, owner3));
+
+        // Verify we can refresh pools for owners based on their products
+        List<String> productIdFromAllOwners = new ArrayList<>();
+        for (Entry<OwnerDTO, Collection<ProductDTO>> entry : ownerToProducts.entrySet()) {
+            List<String> productIds = entry.getValue().stream()
+                .map(p -> p.getId())
+                .collect(Collectors.toList());
+
+            List<AsyncJobStatusDTO> scheduledJobStatuses
+                = productsApi.refreshPoolsForProducts(productIds, true);
+            assertEquals(1, scheduledJobStatuses.size());
+            AsyncJobStatusDTO scheduledJobStatus = scheduledJobStatuses.get(0);
+            verifyRefreshPoolsForProducts(scheduledJobStatus);
+
+            productIdFromAllOwners.add(productIds.get(0));
+        }
+
+        // Verify we refresh pools for each owner
+        List<AsyncJobStatusDTO> scheduledJobStatus
+            = productsApi.refreshPoolsForProducts(productIdFromAllOwners, true);
+        assertEquals(3, scheduledJobStatus.size());
+        verifyRefreshPoolsForProducts(scheduledJobStatus.get(0));
+        verifyRefreshPoolsForProducts(scheduledJobStatus.get(1));
+        verifyRefreshPoolsForProducts(scheduledJobStatus.get(2));
+
+        // Unknown product
+        assertEquals(0, productsApi.refreshPoolsForProducts(Arrays.asList("unknown"), true).size());
+    }
+
+    @Test
+    @DisplayName("throws exception on get_owners with no products")
+    public void getOwnersWithNoProductsException() throws Exception {
+        assertBadRequest(() -> productsApi.getProductOwners(new ArrayList<>()));
+    }
+
+    @Test
+    @DisplayName("throws exception on refresh with no products")
+    public void refreshWithNoProductsException() throws Exception {
+        assertBadRequest(() -> productsApi.refreshPoolsForProducts(new ArrayList<>(), false));
+    }
+
+    @Test
+    @DisplayName("censors owner information on owner-agnostic retrieval")
+    public void ownerAgnosticeRetrievalCensorship() throws Exception {
```

Comment:
It seems to me that the purpose of this test has been lost in time. Looking at the git blame over time, this test used to make sure that fetching a product, the ProductDTO would only show the owner that the user/client that makes the request has access to. But ProductDTO hasn't had the owner as part of the DTO for a long time, so there is no point to this test now. Let's just remove it

Reply:
Removed